nvm.sh

```diff
@@ -704,6 +712,70 @@ nvm_binary_available() {
   nvm_version_greater_than_or_equal_to "$(nvm_strip_iojs_prefix "${1-}")" v0.8.6
 }
 
+nvm_set_colors() {
+  if nvm_has_colors; then
+    local INSTALLED_COLOR
+    local LTS_AND_SYSTEM_COLOR
+    local CURRENT_COLOR
+    local NOT_INSTALLED_COLOR
+    local DEFAULT_COLOR
+
+    INSTALLED_COLOR="$(echo "$1" | awk '{ print substr($0, 1, 1); }')"
+    LTS_AND_SYSTEM_COLOR="$(echo "$1" | awk '{ print substr($0, 2, 1); }')"
+    CURRENT_COLOR="$(echo "$1" | awk '{ print substr($0, 3, 1); }')"
+    NOT_INSTALLED_COLOR="$(echo "$1" | awk '{ print substr($0, 4, 1); }')"
+    DEFAULT_COLOR="$(echo "$1" | awk '{ print substr($0, 5, 1); }')"
+    nvm_echo_with_colors "Setting colors to: \033[$(nvm_print_color_code "${INSTALLED_COLOR}") ${INSTALLED_COLOR}\033[$(nvm_print_color_code "${LTS_AND_SYSTEM_COLOR}") ${LTS_AND_SYSTEM_COLOR}\033[$(nvm_print_color_code "${CURRENT_COLOR}") ${CURRENT_COLOR}\033[$(nvm_print_color_code "${NOT_INSTALLED_COLOR}") ${NOT_INSTALLED_COLOR}\033[$(nvm_print_color_code "${DEFAULT_COLOR}") ${DEFAULT_COLOR}\033[0m"
```

Comment:
one thing i notice is that if you call `nvm_set_colors` with invalid colors, `nvm_print_color_code` will error and print out "invalid color code", but `nvm_set_colors` both won't fail, and will also return zero.

Can we find a way to validate `$1` before getting to the extraction?

Reply:
In the case statement checking for the `--set-colors` flag, there is an if statement doing this validation before `nvm_set_colors` gets called:
```
...
    "--set-colors")
      if [ "${#1}" -eq 5 ] && nvm_echo "$1" | nvm_grep -E "^[:rRgGbBcCyYmMkKeW:]{1,}$" 1>/dev/null; then
        nvm_set_colors "$1"
      ...
```

Is it preferable to put this validation within `nvm_set_colors` rather than in the code that calls the function?

Reply:
Yes, I think that'd be ideal - it'd help unit testing as well.

Reply:
Ok, moving the validation code to `nvm_set_colors`.